File: yt-transcript-app/frontend/src/types/video.ts
export interface Video {
  id: string;
  title: string;
  thumbnail: string | null;
  upload_date: string | null;
  duration: number | null;
  has_transcript: boolean;
}

export interface TranscriptLine {
  start: number;
  end: number;
  text: string;
}

export interface VideoListResponse {
  videos: Video[];
  last_updated: string | null;
}

export interface TranscriptResponse {
  video_id: string;
  title: string;
  lines: TranscriptLine[];
}

File: yt-transcript-app/backend/app/schemas/video.py
from pydantic import BaseModel
from typing import List, Optional
from datetime import datetime


class TranscriptLine(BaseModel):
    start: float
    end: float
    text: str


class Video(BaseModel):
    id: str
    title: str
    thumbnail: Optional[str] = None
    upload_date: Optional[str] = None
    duration: Optional[int] = None
    has_transcript: bool = False


class VideoListResponse(BaseModel):
    videos: List[Video]
    last_updated: Optional[datetime] = None


class TranscriptResponse(BaseModel):
    video_id: str
    title: str
    lines: List[TranscriptLine]